### PR TITLE
Update Microsoft.CodeAnalysis.CSharp.CodeStyle version

### PR DESCRIPTION
**Pull request by stephentoub**
Also updates our config files to make them more easily diffable with the ones that come in the analyzer nuget package, and moves some of the rules from silent (meaning they only show up if you click on the line in VS) to suggestion (meaning they show up as suggestions in VS but have no impact on builds).

### CODE REVIEW

File: eng/CodeAnalysis.src.globalconfig [hunk 1]
```diff
@@ -1257,293 +1257,278 @@ dotnet_diagnostic.SA1643.severity = none
...
 dotnet_diagnostic.IDE0078.severity = silent
 
 # IDE0079: RemoveUnnecessarySuppression
 dotnet_diagnostic.IDE0079.severity = silent
 
-# IDE0080: RemoveConfusingSuppressionForIsExpression
+# IDE0080: Remove unnecessary suppression operator
 dotnet_diagnostic.IDE0080.severity = silent
 
 # IDE0081: RemoveUnnecessaryByVal
```
Comment (danmoseley) [inline eng/CodeAnalysis.src.globalconfig:1476]: these names don't have spaces?

File: eng/CodeAnalysis.src.globalconfig [hunk 2]
```diff
@@ -1257,293 +1257,278 @@ dotnet_diagnostic.SA1643.severity = none
...
-# IDE2001: EmbeddedStatementsMustBeOnTheirOwnLine
+# IDE2001: Embedded statements must be on their own line
 dotnet_diagnostic.IDE2001.severity = silent
 
-# IDE2002: ConsecutiveBracesMustNotHaveBlankLinesBetweenThem
+# IDE2002: Consecutive braces must not have blank line between them
 dotnet_diagnostic.IDE2002.severity = silent
 
-# IDE2003: ConsecutiveStatementPlacement
+# IDE2003: C#
```
Comment (danmoseley) [inline eng/CodeAnalysis.src.globalconfig:1530]: that's the name? c#?

**Review by danmoseley — APPROVED** [review]

File: eng/CodeAnalysis.src.globalconfig (hunk 2 above)
Comment (stephentoub) [reply eng/CodeAnalysis.src.globalconfig:1530]: That's what's in the nuget config file

Comment (danmoseley) [reply eng/CodeAnalysis.src.globalconfig:1530]: my care level is low but cc @sharwell

Comment (sharwell) [reply eng/CodeAnalysis.src.globalconfig:1530]: @mavasani lol

File: eng/Versions.props [hunk 3]
```diff
@@ -43,13 +43,13 @@
...
-    <MicrosoftCodeAnalysisCSharpWorkspacesVersion>4.0.0-3.final</MicrosoftCodeAnalysisCSharpWorkspacesVersion>
-    <MicrosoftCodeAnalysisVersion>4.0.0-3.final</MicrosoftCodeAnalysisVersion>
+    <MicrosoftCodeAnalysisCSharpWorkspacesVersion>4.0.0-4.final</MicrosoftCodeAnalysisCSharpWorkspacesVersion>
+    <MicrosoftCodeAnalysisVersion>4.0.0-4.final</MicrosoftCodeAnalysisVersion>
   </PropertyGroup>
   <PropertyGroup>
     <!-- Code analysis dependencies -->
-    <MicrosoftCodeAnalysisCSharpCodeStyleVersion>3.10.0</MicrosoftCodeAnalysisCSharpCodeStyleVersion>
-    <MicrosoftCodeAnalysisCSharpVersion>3.10.0</MicrosoftCodeAnalysisCSharpVersion>
+    <MicrosoftCodeAnalysisCSharpCodeStyleVersion>4.0.0-4.final</MicrosoftCodeAnalysisCSharpCodeStyleVersion>
```
Comment (ViktorHofer) [inline eng/Versions.props:51]: @sharwell which version of VS is required when using the updated version? I assume 2022?

Comment (stephentoub) [reply eng/Versions.props:51]: I assume so.

Comment (stephentoub) [reply eng/Versions.props:51]: @ViktorHofer, do you have concerns?

Comment (sharwell) [reply eng/Versions.props:51]: > which version of VS is required when using the updated version?

2022 Preview 4

Comment (ViktorHofer) [reply eng/Versions.props:51]: The docs list VS 2019 as a supported version: https://github.com/dotnet/runtime/blob/main/docs/workflow/requirements/windows-requirements.md#visual-studio. Would make sense to update them.

Comment (stephentoub) [reply eng/Versions.props:51]: A few lines earlier we're changing several 4.0.0-3.final versions to 4.0.0-4.final.  Isn't 4.0.0-3.final also VS2022, just Preview 3?

Regardless, yes, I can update the Windows requirements.  Folks should be using VS2022.  We should make sure that we never require a preview that hasn't been released yet, but I think beyond that is fair game.

Comment (ViktorHofer) [reply eng/Versions.props:51]: We probably already require VS 2022 but just forgot to update the docs. Agreed that we shouldn't depend on a VS version not yet released but preview is fine.

**Pull request merged**